providers/implementations/ciphers/cipher_sm4_gcm_hw.c

```diff
@@ -91,6 +91,8 @@ static const PROV_GCM_HW sm4_gcm = {
 
 #if defined(OPENSSL_CPUID_OBJ) && defined(__riscv) && __riscv_xlen == 64
 # include "cipher_sm4_gcm_hw_rv64i.inc"
+#elif defined(OPENSSL_CPUID_OBJ) && (defined(__x86_64) || defined(__x86_64__) || defined(_M_X64))
```

Comment:
Checks for `_M_AMD64` appear throughout the OpenSSL code.
Wondering if we should add it here too.